ksql-engine/src/test/java/io/confluent/ksql/physical/PhysicalPlanBuilderTest.java

```diff
@@ -105,4 +118,107 @@ public void shouldCreateExecutionPlan() throws Exception {
     Assert.assertEquals(lines[5], "\t\t\t\t\t\t\t\t\t\t > [ SOURCE ] Schema: [TEST1.COL0 : INT64 , TEST1.COL1 : STRING , TEST1.COL2 : STRING , TEST1.COL3 : FLOAT64 , TEST1.COL4 : ARRAY , TEST1.COL5 : MAP].");
   }
 
+  @Test
+  public void shouldAddMetricsInterceptors() throws Exception {
+    final QueryMetadata queryMetadata = buildPhysicalPlan(simpleSelectFilter);
+    StreamsConfig config = queryMetadata.getStreamsConfig();
+
+    Object val = config.originals().get(
+        StreamsConfig.consumerPrefix(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG));
+    Assert.assertTrue(val instanceof List);
+    List<String> consumerInterceptors = (List<String>) val;
+    Assert.assertEquals(1, consumerInterceptors.size());
+    Assert.assertEquals(ConsumerCollector.class, Class.forName(consumerInterceptors.get(0)));
+
+    val = config.originals().get(
+        StreamsConfig.producerPrefix(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG));
+    Assert.assertTrue(val instanceof List);
+    List<String> producerInterceptors = (List<String>) val;
+    Assert.assertEquals(1, producerInterceptors.size());
```

Comment:
next 2 lines can be replaced with `assertThat(expectedList, equalTo(actualList))`

Reply:
Had something like that originally. Felt like the current check was more robust. We're checking that the name passed to the class loader results in our interceptor class being loaded.